pkg/providers/tinkerbell/validator.go

```diff
@@ -0,0 +1,152 @@
+		return err
+	}
+
+	if err := v.validatetinkerbellPBnJGRPCAuth(ctx, datacenterConfig.Spec.TinkerbellPBnJGRPCAuth); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// TODO: dry out machine configs validations
+func (v *Validator) ValidateClusterMachineConfigs(ctx context.Context, tinkerbellClusterSpec *spec) error {
+	// TODO: move this to api Cluster validations
+	if len(tinkerbellClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host) <= 0 {
+		return errors.New("cluster controlPlaneConfiguration.Endpoint.Host is not set or is empty")
+	}
+
+	if tinkerbellClusterSpec.Cluster.Spec.ControlPlaneConfiguration.MachineGroupRef == nil {
+		return errors.New("must specify machineGroupRef for control plane")
+	}
+
+	controlPlaneMachineConfig := tinkerbellClusterSpec.controlPlaneMachineConfig()
+	if controlPlaneMachineConfig == nil {
+		return fmt.Errorf("cannot find tinkerbellClusterSpec %v for control plane", tinkerbellClusterSpec.Cluster.Spec.ControlPlaneConfiguration.MachineGroupRef.Name)
+	}
+	if len(controlPlaneMachineConfig.Spec.OSFamily) <= 0 {
+		return errors.New("tinkerbellMachineConfig osFamily for control plane is not set or is empty")
+	}
+
+	if tinkerbellClusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations[0].MachineGroupRef == nil {
+		return errors.New("must specify machineGroupRef for worker nodes")
+	}
+
+	workerNodeGroupMachineConfig := tinkerbellClusterSpec.firstWorkerMachineConfig()
+	if workerNodeGroupMachineConfig == nil {
+		return fmt.Errorf("cannot find tinkerbellMachineConfig %v for worker nodes", tinkerbellClusterSpec.Cluster.Spec.WorkerNodeGroupConfigurations[0].MachineGroupRef.Name)
+	}
+	if len(workerNodeGroupMachineConfig.Spec.OSFamily) <= 0 {
+		return errors.New("tinkerbellMachineConfig osFamily for worker nodes is not set or is empty")
+	}
+
+	// TODO: move this to api Cluster validations
+	controlPlaneEndpointIp := tinkerbellClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host
+	if controlPlaneEndpointIp == "" {
+		return fmt.Errorf("controlPlaneConfiguration.Endpoint.Host is required")
+	}
+	if err := v.validateIP(tinkerbellClusterSpec.Cluster.Spec.ControlPlaneConfiguration.Endpoint.Host); err != nil {
+		return fmt.Errorf("cluster controlPlaneConfiguration.Endpoint.Host is invalid: %s", controlPlaneEndpointIp)
+	}
+
+	if controlPlaneMachineConfig.Spec.OSFamily != anywherev1.Bottlerocket && controlPlaneMachineConfig.Spec.OSFamily != anywherev1.Ubuntu {
+		return fmt.Errorf("control plane osFamily: %s is not supported, please use one of the following: %s, %s", controlPlaneMachineConfig.Spec.OSFamily, anywherev1.Bottlerocket, anywherev1.Ubuntu)
+	}
+
+	if workerNodeGroupMachineConfig.Spec.OSFamily != anywherev1.Bottlerocket && workerNodeGroupMachineConfig.Spec.OSFamily != anywherev1.Ubuntu {
+		return fmt.Errorf("worker node osFamily: %s is not supported, please use one of the following: %s, %s", workerNodeGroupMachineConfig.Spec.OSFamily, anywherev1.Bottlerocket, anywherev1.Ubuntu)
+	}
```

Comment:
can we make it Ubuntu only for now? We can add bottlerocket in here as well once we support it